feat: Use /v1/chat/completions for OpenAI code-completions

PhilKes opened the pull request:
Fixes #369.

This PR changes the Code-Completion for `OpenAI` ServiceType to use `/v1/chat/completions` instead of `/v1/completions`.
There is no `suffix` parameter for OpenAI Chat-Completions, therefore the FIM prompt template is used instead as a single "user" message.

Unfortunately the results dont look right to me just yet, I'm not sure why, here's an example: 
![image](https://github.com/carlrobertoh/CodeGPT/assets/39240633/c62a4132-9945-4d23-8dcf-c164abb10f73)

reneleonhardt commented:
@PhilKes Thank you for providing a fix, I rebased for you and converted your code to Kotlin.
`OpenAISettingsState` doesn't contain isUsingCustomPath and path anymore, I commented it out.

Can you check if the current code fixes the problem you have seen? 
https://github.com/reneleonhardt/CodeGPT/tree/openai-code-use-chat-completions

PhilKes commented:
> @PhilKes Thank you for providing an implementation, I rebased for you and converted your code to Kotlin. `OpenAISettingsState` doesn't contain `isUsingCustomPath` and `path` anymore, I commented it out.
> 
> Can you check if the current code fixes the problem you have seen? https://github.com/reneleonhardt/CodeGPT/tree/openai-code-use-chat-completions

I am not sure if this PR is still relevant, I haven't been working on CodeGPT for some weeks now.
See https://github.com/carlrobertoh/CodeGPT/issues/369#issuecomment-1951444495
Perhaps this can be closed @carlrobertoh ?

adam6806 commented:
Can I ask why "Code completions are only available with gpt-3.5-turbo-instruct model"? This model is much more expensive than say 4o-mini. Is it not possible to get code completions to work with that model?
